shared/modules/selectors/networks.ts

```diff
@@ -0,0 +1,114 @@
+import {
+  RpcEndpointType,
+  type NetworkConfiguration,
+  type NetworkState as _NetworkState,
+} from '@metamask/network-controller';
+import { createSelector } from 'reselect';
+import { NetworkStatus } from '../../constants/network';
+import { createDeepEqualSelector } from './util';
+
+export type NetworkState = { metamask: _NetworkState };
```

Comment:
Whatever we do, it will change again soon when we unflatten the `metamask` state, so we can re-evalute what we want to do then. All of the background state types we use in the UI are a mess right now.